File: 프로그래머스/lv2/64065. 튜플/튜플.py
def solution(s):
    l1 = s.split("}")
    answer = []
    
    for i in range(len(l1)):
        l1[i] = l1[i].replace('{', '')
        l1[i] = l1[i].strip(',')
    l1 = ' '.join(l1).split()
    l2 = [[0]*500 for _ in range(500)] 
    
    for i in range(len(l1)):
        l2[i] = l1[i].split(',')
    for i in range(len(l1)):
        for j in range(len(l2[i])):
            l2[i][j] = int(l2[i][j])
            
    for i in range(1, len(l1)+1):
        for j in l2:
            if len(j) == i:
                for k in j:
                    if k not in answer:
                        answer.append(k)

    return answer

File: 프로그래머스/2/64065. 튜플/튜플.java
import java.util.*;

/*
맨 왼쪽 괄호 삭제, 맨 오른쪽 괄호 삭제
},{ 로 split,, String 배열에 넣기
*/

class Solution {
    public int[] solution(String s) {
        
        ArrayList<Integer> l1 = new ArrayList<>();
        HashSet<Integer> hs = new HashSet<>();
        
        s = s.substring(2, s.length()-2);
        String[] a = s.split("},\\{");
        
        Arrays.sort(a, (o1, o2) -> {
            return o1.length() - o2.length();
        });
        
        for (int i = 0; i < a.length; i++) {
            String[] tmp = a[i].split(",");
            
            for (int j = 0; j < tmp.length; j++) {
                if (hs.contains(Integer.parseInt(tmp[j]))) continue;
                
                hs.add(Integer.parseInt(tmp[j]));
                l1.add(Integer.parseInt(tmp[j]));
            }
        }
        
        int[] answer = new int[l1.size()];
        
        for (int i = 0; i < l1.size(); i++) {
            answer[i] = l1.get(i);
        }
        
        return answer;
    }
}